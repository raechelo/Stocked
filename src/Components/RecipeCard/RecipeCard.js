import React from 'react'
import youtube from '../../assets/youtube.svg';
import Instructions from '../Instructions/Instructions';

export default function RecipeCard(props) {
  const { name, category, image, link, tags, yt } = props
  return (
    <article className="Recipe-Card">
      <h5>{name}</h5>
      <section>
        <p><span>{props.measure1}</span>{props.ingredient1}</p>
        <p><span>{props.measure2}</span>{props.ingredient2}</p>
        <p><span>{props.measure3}</span>{props.ingredient3}</p>
        <p><span>{props.measure4}</span>{props.ingredient4}</p>
        <p><span>{props.measure5}</span>{props.ingredient5}</p>
        <p><span>{props.measure6}</span>{props.ingredient6}</p>
        <p><span>{props.measure7}</span>{props.ingredient7}</p>
        <p><span>{props.measure8}</span>{props.ingredient8}</p>
        <p><span>{props.measure9}</span>{props.ingredient9}</p>
        <p><span>{props.measure10}</span>{props.ingredient10}</p>
        <p><span>{props.measure11}</span>{props.ingredient11}</p>
        <p><span>{props.measure12}</span>{props.ingredient12}</p>
        <p><span>{props.measure13}</span>{props.ingredient13}</p>
        <p><span>{props.measure14}</span>{props.ingredient14}</p>
        <p><span>{props.measure15}</span>{props.ingredient15}</p>
        <p><span>{props.measure16}</span>{props.ingredient16}</p>
        <p><span>{props.measure17}</span>{props.ingredient17}</p>
        <p><span>{props.measure18}</span>{props.ingredient18}</p>
        <p><span>{props.measure19}</span>{props.ingredient19}</p>
        <p><span>{props.measure20}</span>{props.ingredient20}</p>
      </section>
      <p className="instr">{Instructions}</p>
      <a className="link" href={link}>{link}</a>
      <a className="yt-link" href={yt}><img src={youtube}/></a>
      {tags.split(',').join(' ').split(' ').map(i => (`#${i}`))}
      <h6>{category}</h6>
      <img src={image} alt={`${name} picture`} />
    </article>
  )
}
