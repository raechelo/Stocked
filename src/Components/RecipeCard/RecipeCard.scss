.Recipe-Card {
  background-color: $card-bg;
  border: 2px solid $card-border;
  border-radius: 5px;
  display: grid;
  font-family: $body-font;
  grid-template-areas: "name name img"
  "ing ing img"
  "inst inst inst"
  "link yt .";
  grid-template-columns: repeat(3, 1fr);
  margin-top: 1rem;
  padding: 1rem;
  width: 40%;

  h5 {
    font-family: $fancy-font;
    font-size: 1.5rem;
    grid-area: name;
    margin: 0;
  }

  section {
    grid-area: ing;
  }

  .instr {
    grid-area: inst;
    max-height: 4rem;
    overflow: scroll;
  }

  .link {
    text-decoration: underline;
    grid-area: link;
    margin: auto;
    &:hover {
      color: $fave;
      cursor: pointer;
    }
  }

  .yt-link {
    cursor: pointer;
    grid-area: yt;
    width: 2.5rem;
    margin: auto;
  }

  .tags {
    color: $fave;
    font-weight: bolder;
    margin: auto;
  }

  .meal-pic {
    grid-area: img;
    width: 100%;
  }

  section p {

    margin: 0;

    span {
      font-weight: bolder;
      margin: 0 .5rem;
    }
  }
}