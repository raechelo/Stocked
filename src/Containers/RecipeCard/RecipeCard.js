import React from 'react'

export default function RecipeCard(props) {
  const { strMeal } = props
  return (
    <div>
      <h5>{strMeal}</h5>
    </div>
  )
}
