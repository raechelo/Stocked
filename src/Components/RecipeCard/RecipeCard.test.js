import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from './RecipeCard';

describe('RecipeCard', () => {

  let wrapper;
  
  let mockRecipe = {
    name: 'Mac & Cheese',
    measure1: '1 cup',
    measure2: '1 cup',
    ingredient1: 'mac',
    ingredient2: 'cheese',
    instructions: 'boil water, cook noods. add cheese. eat.',
    link: 'www.macandcheese.com',
    yt: 'www.youtube.com/macandcheese',
    tags: 'noods, cheese',
    category: 'american',
    image: 'www.macandcheese.com/image'
  }

  beforeEach(() => {
    wrapper = shallow(<RecipeCard recipe={mockRecipe}/>);
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should split the tags and return with octothorps', () => {

    mockRecipe = {
      name: 'Mac & Cheese',
      measure1: '1 cup',
      measure2: '1 cup',
      ingredient1: 'mac',
      ingredient2: 'cheese',
      instructions: 'boil water, cook noods. add cheese. eat.',
      link: 'www.macandcheese.com',
      yt: 'www.youtube.com/macandcheese',
      category: 'american',
      image: 'www.macandcheese.com/image'
    }

    expect(wrapper).toMatchSnapshot();
  });
});