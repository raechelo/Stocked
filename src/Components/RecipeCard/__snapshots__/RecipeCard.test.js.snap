// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match the snapshot with all the data passed in 1`] = `ShallowWrapper {}`;

exports[`RecipeCard should split the tags and return with octothorps 1`] = `ShallowWrapper {}`;
